File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.length() == 0)
            return 0;
        int count = 1;
        unordered_set<char> st;
        int i = 0, j = 0;
        while(j<s.length()){
            if(st.find(s[j]) != st.end()){
                count = max(count, j-i);
                while(s[i]!=s[j]){
                    st.erase(s[i++]);
                }
                i++;
            }else{
                st.insert(s[j]);
            }
            j++;
        }
        count = max(count, j-i);
        return count;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int longestSubstring = 0;
        int i = 0, j = 0;
        Set<Character> st = new HashSet<>();
        while(j<s.length()){
            if(st.contains(s.charAt(j))){
                while(st.contains(s.charAt(j))){
                    st.remove(s.charAt(i));
                    i++;
                }
            }
            st.add(s.charAt(j));
            j++;
        
            longestSubstring = Math.max(longestSubstring, st.size());
        }
        return longestSubstring;
    }
}